Union Find/unionFind.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Matheus Oliveira
+ * 29/10/2017
+ * unionFind.cpp
+ * 
+ * Algorithm of Union Find optimized using Dynamic Programming and Rank of sets.
+ * 
+*/
+
+#include <iostream>
+#define MAXN 1000000
+
+int rank[MAXN], parents[MAXN];
+
+void initialize(int parents[], int rank[], int n) {
```

Comment:
Do you initializes the full arrays ```rank``` and ```parents``` ?
  * If so then the function don't need parameter.  

Or do you only initializes parts of them?